include/mockturtle/algorithms/cut_enumeration.hpp

```diff
@@ -604,4 +606,256 @@ network_cuts<Ntk, ComputeTruth, CutData> cut_enumeration( Ntk const& ntk, cut_en
   return res;
 }
 
+// This function expects to receive a network where nodes are sorted in
+// topological order. Cuts are represented as a 64-bit bit vector where each bit
+// determines whether a given node exists in the cut.
+
+/*! \brief Cut enumeration.
+ *
+ * This function implements a generic fast cut enumeration algorithm for graphs
+ * containing at most 64 nodes. It is generic as it supports graphs in which
+ * nodes can have variable fan-in. Speed and space-efficiency are achieved by
+ * representing cuts as 64-bit bit vectors, i.e. each bit represents whether or
+ * not a node is in a cut. Cut-set union and domination operations then
+ * transform into bitwise operations that can be performed in a few clock cycles
+ * each.
+ *
+ * Like the larger cut_enumeration algorithm, this algorithm traverses all nodes
+ * in topological order and computes a node's cuts based on its fanins' cuts.
+ * Dominated cuts are filtered and are not added to the cut set. For each node a
+ * unit cut is added to the end of each cut set.
+ *
+ * This function computes all cuts of the network (i.e. the number of generated
+ * cuts is not bounded). Though the number of cuts cannot be bounded, their size
+ * can be bound by passing a `cut_size` argument to the function.
+ *
+ * **Required network functions:**
+ * - `fanin_size`
+ * - `foreach_fanin`
+ * - `foreach_gate`
+ * - `foreach_pi`
+ * - `get_node`
+ * - `node_to_index`
+ * - `size`
+ *
+ * Note that this algorithm *only* works for graphs with at most 64 nodes.
+ * However, since we cannot know the size of a graph at compile-time, this
+ * function returns a boolean value in addition to the cut sets to indicate
+ * whether the input graph is valid or not, and therefore whether the cuts are
+ * to be interpreted by the calling program.
+ *
+ * \verbatim embed:rst
+ *
+ * .. warning::
+ *
+ *    This algorithm expects the nodes in the network to be in topological
+ *    order.  If the network does not guarantee a topological order of nodes one
+ *    can wrap the network parameter in a ``topo_view`` view.
+ */
+template<typename Ntk>
+std::pair<bool, std::vector<std::vector<uint64_t>>>
```

Comment:
An alternative return type is `std::optional<std::vector<std::vector<uint64_t>>>`